### ISSUE TITLE
Partial fit support

### ISSUE THREAD

**Issue by Vichoko** [ldeecke/gmm-torch#7]
In previous issues, #2 and #4 have been said that currently `fit `method reset the parameters on each call, so it isn't possible to batch or epoch training the same models with different sets of data points.

I'm not gonna insist on the utility of this, just noting that modern datasets don't fit in memory completely and representative sampling isn't really viable in most cases due to the high dimensionality and quantity of data points.

I want to know if this feature is viable or if it is possible for a GMM to be fitted in rounds with different data points, just like `partial_fit` or a gradient descent step is made in other architectures.

I know Sklearn's GMM doesn't have partial_fit either, but why?

Is it planned for the future to implement this?

**Comment by ldeecke** [comment]
Hi,

This question is somewhat beyond the scope of this repository. For a discussion, see [this](https://stackoverflow.com/questions/29095769/sklearn-gmm-on-large-datasets) post on Stack Overflow; [this](https://stats.stackexchange.com/questions/64193/fitting-a-gaussian-mixture-model-using-stochastic-gradient-descent) post on Cross Validated.

**Comment by HongLouyemeng** [comment]
> @yangzhaopeng3 Well, this was a huge limitation for my thesis research, but also important to note that was vital to mention these limitations on any research that's made while using this model. Besides this, I'd also consider comparing if the GMM with its limitations performs better or worse than a model which can be trained in batches. It can happen that a GMM with a smaller but more feature-engineered dataset can perform better than a NN with a bigger raw dataset.
> 
> In the end, I had to accept these memory limitations and started looking for computer resources in Universities. I ended up getting access to a supercomputer with +256 GB of RAM which enabled me to train a Sklearn's GMM in some weeks with a huge dataset. My NN-based model ended up getting better metrics though.

Money makes you stronger
